sdk/src/trace/tracer_provider.cc

```diff
@@ -23,6 +23,9 @@ opentelemetry::nostd::shared_ptr<opentelemetry::trace::Tracer> TracerProvider::G
     nostd::string_view library_name,
     nostd::string_view library_version) noexcept
 {
+  tracer_->SetInstrumentationLibrary(
+      opentelemetry::instrumentationlibrary::InstrumentationLibrary::create(library_name,
```

Comment:
Each tracer is identified by `library_name`. And `TracerProvider` should create separate instances of tracer for each library_name as input to `TracerProvider::GetTracer`. And further returns the same tracer if it already exists for input `library_name`. Default (non-no-op) tracer should be created if the `library_name` is empty.  This can be achieved by `TracerProvider` maintaining a list of tracers (mapped with their `library_name`).

Though not related to this PR. Will create a separate ticket for that.

Reply:
Yes, I was thinking on this but thought it may be good to do it in a separate PR, as currently `TracerProvider` contains only 1 `Tracer`. Opened below ticket and I'll look into it.

https://github.com/open-telemetry/opentelemetry-cpp/issues/694

Reply:
+1 to solving this.